tests/chart_tests/test_ingress.py

```diff
@@ -32,10 +32,66 @@ def test_basic_ingress(self, kube_version):
+        {"host":"registry.example.com","http":{"paths":[{"path":"/","pathType":"Prefix","backend":{"service":{"name":"release-name-registry","port":{"name":"registry-http"}}}}]}}]
         """
         )
 
         assert doc["apiVersion"] == "networking.k8s.io/v1"
         assert doc["spec"]["rules"] == expected_rules_v1
+
+    def test_astro_ui_per_host_ingress(self, kube_version):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"global": {"enablePerHostIngress": True}},
+            show_only=[
+                "charts/astronomer/templates/astro-ui/astro-ui-ingress.yaml",
+                "charts/astronomer/templates/ingress.yaml",
+            ],
+        )
+        assert len(docs) == 2
+        expected_astroui_rules_v1 = json.loads(
+            """
+        [
+            {"host":"app.example.com","http":{"paths":[{"path":"/","pathType":"Prefix","backend":{"service":{"name":"release-name-astro-ui","port":{"name":"astro-ui-http"}}}}]}}
+        ]
+        """
+        )
+        assert docs[0]["apiVersion"] == "networking.k8s.io/v1"
+        assert docs[0]["spec"]["rules"] == expected_astroui_rules_v1
+
+        expected_common_rules_v1 = json.loads(
+            """
+        [
+            {"host":"example.com","http":{"paths":[{"path":"/","pathType":"Prefix","backend":{"service":{"name":"release-name-astro-ui","port":{"name":"astro-ui-http"}}}}]}}
+        ]
+        """
+        )
+        assert docs[1]["apiVersion"] == "networking.k8s.io/v1"
+        assert docs[1]["spec"]["rules"] == expected_common_rules_v1
+
+    def test_registry_per_host_ingress(self, kube_version):
+        docs = render_chart(
+            kube_version=kube_version,
+            values={"global": {"enablePerHostIngress": True}},
+            show_only=[
+                "charts/astronomer/templates/registry/registry-ingress.yaml",
+                "charts/astronomer/templates/ingress.yaml",
+            ],
+        )
+        assert len(docs) == 1
+        expected_rules_v1 = json.loads(
+            """
+        [
+        {"host":"registry.example.com","http":{"paths":[{"path":"/","pathType":"Prefix","backend":{"service":{"name":"release-name-registry","port":{"name":"registry-http"}}}}]}}]
+        """
+        )
+        assert docs[0]["apiVersion"] == "networking.k8s.io/v1"
+        assert docs[0]["spec"]["rules"] == expected_rules_v1
+
+    def test_single_ingress_per_host(self, kube_version):
+        default_docs = render_chart(values={"global": {"enablePerHostIngress": True}})
+        ingresses = [
+            doc for doc in default_docs if doc["kind"].lower() == "Ingress".lower()
+        ]
+        assert all(len(i["spec"]["rules"]) == 1 for i in ingresses)
+        assert all(len(i["spec"]["tls"][0]["hosts"]) == 1 for i in ingresses)
```

Comment:
I think that rather than having several tests functions per-ingress, we could have one test or one parametrized test that pulls all ingresses from the chart and tests them. That's an optimization though, so if we want to not do it and spend that time on something else then that's fine.